batch/extension-jberet/src/main/java/org/wildfly/extension/batch/jberet/deployment/SecurityAwareBatchEnvironment.java

```diff
@@ -0,0 +1,91 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.wildfly.extension.batch.jberet.deployment;
+
+import org.jberet.spi.BatchEnvironment;
+import org.wildfly.common.Assert;
+import org.wildfly.security.auth.server.RealmUnavailableException;
+import org.wildfly.security.auth.server.SecurityDomain;
+import org.wildfly.security.auth.server.SecurityIdentity;
+import org.wildfly.security.auth.server.ServerAuthenticationContext;
+
+/**
+ * A batch environment which can provide various security options.
+ *
+ * @author <a href="mailto:jperkins@redhat.com">James R. Perkins</a>
+ */
+public interface SecurityAwareBatchEnvironment extends BatchEnvironment {
+
+    /**
+     * Returns the security domain if defined or {@code null} if not defined.
+     *
+     * @return the security domain or {@code null} if not defined
+     */
+    SecurityDomain getSecurityDomain();
+
+    /**
+     * If the {@linkplain #getSecurityDomain() security domain} is not {@code null} the current user is returned.
+     * otherwise {@code null} is returned.
+     * <p>
+     * Note that if the current identity is anonymous {@code null} will be returned.
+     * </p>
+     *
+     * @return the current user name or {@code null}
+     */
+    default String getCurrentUserName() {
+        final SecurityIdentity securityIdentity = getIdentity();
+        if (securityIdentity != null && !securityIdentity.isAnonymous()) {
+            return securityIdentity.getPrincipal().getName();
+        }
+        return null;
+    }
+
+    /**
+     * If the {@linkplain #getSecurityDomain() security domain} is not {@code null} the current identity is returned.
+     * otherwise {@code null} is returned.
+     *
+     * @return the current identity or {@code null}
+     */
+    default SecurityIdentity getIdentity() {
+        final SecurityDomain securityDomain = getSecurityDomain();
+        if (securityDomain != null) {
+            return securityDomain.getCurrentSecurityIdentity();
+        }
+        return null;
+    }
+
+    /**
+     * Creates an identity if the current {@linkplain #getSecurityDomain() security domain} is not {@code null},
+     * otherwise {@code null} is returned
+     *
+     * @param user the user to attempt to create an identity for, cannot be {@code null}
+     *
+     * @return the created identity or {@code null}
+     *
+     * @throws RealmUnavailableException if the realm is not available
+     */
+    default SecurityIdentity createIdentity(final String user) throws RealmUnavailableException {
+        final SecurityDomain securityDomain = getSecurityDomain();
+        if (securityDomain != null) {
+            ServerAuthenticationContext authenticationContext = securityDomain.createNewAuthenticationContext();
+            authenticationContext.setAuthenticationName(Assert.checkNotEmptyParam("user", user));
+            authenticationContext.authorize();
```

Comment:
Just need to discuss this specific line with the Elytron team as this will cause the LoginPermission to be required which I don't think makes sense in this context.